lurk-fibonacci/benches/throughput.rs

```diff
@@ -1,50 +1,36 @@
-use std::{cell::RefCell, rc::Rc, sync::Arc, time::Duration};
-
 use anyhow::anyhow;
+use camino::Utf8PathBuf;
 use criterion::{
     black_box, criterion_group, criterion_main, measurement, BatchSize, BenchmarkGroup,
     BenchmarkId, Criterion, SamplingMode,
 };
-
+use once_cell::sync::Lazy;
 use pasta_curves::pallas;
+use std::{sync::Arc, time::Duration};
+
+use lurk_fibonacci::{fib_expr, fib_frame, fib_limit};
 
 use lurk::{
+    cli::paths::lurk_default_dir,
+    config::lurk_config,
     eval::lang::{Coproc, Lang},
-    field::LurkField,
-    lem::{eval::evaluate, multiframe::MultiFrame, pointers::Ptr, store::Store},
+    lem::{eval::evaluate, multiframe::MultiFrame, store::Store},
     proof::nova::NovaProver,
     proof::Prover,
     public_parameters::{
         instance::{Instance, Kind},
         public_params,
     },
-    state::State,
 };
 
-mod common;
-use common::set_bench_config;
+/// Edit this path to use a config file specific to benchmarking
+/// E.g. `Utf8PathBuf::from("/home/<user>/lurk-rs/lurk-bench.toml");`
+pub(crate) static BENCH_CONFIG_PATH: Lazy<Utf8PathBuf> =
+    Lazy::new(|| lurk_default_dir().join("lurk.toml"));
 
-fn fib<F: LurkField>(store: &Store<F>, state: Rc<RefCell<State>>, _a: u64) -> Ptr {
-    let program = r#"
-(letrec ((next (lambda (a b) (next b (+ a b))))
-           (fib (next 0 1)))
-  (fib))
-"#;
-
-    store.read(state, program).unwrap()
-}
-
-// The env output in the `fib_frame`th frame of the above, infinite Fibonacci computation will contain a binding of the
-// nth Fibonacci number to `a`.
-// means of computing it.]
-fn fib_frame(n: usize) -> usize {
-    11 + 10 * n
-}
-
-// Set the limit so the last step will be filled exactly, since Lurk currently only pads terminal/error continuations.
-fn fib_limit(n: usize, rc: usize) -> usize {
-    let frame = fib_frame(n);
-    rc * (frame / rc + usize::from(frame % rc != 0))
+/// Sets the config settings with the given file
+pub(crate) fn set_bench_config() {
```

Comment:
This is already in `./benches/common`. Can we avoid copying it?

Reply:
Moving instead of copying

Reply:
No